mmpose/datasets/datasets/mesh/mosh_dataset.py

```diff
@@ -0,0 +1,66 @@
+import copy as cp
+from abc import ABCMeta
+
+import numpy as np
+from torch.utils.data import Dataset
+
+from mmpose.datasets.builder import DATASETS
+from mmpose.datasets.pipelines import Compose
+
+
+@DATASETS.register_module()
+class MoshDataset(Dataset, metaclass=ABCMeta):
+    """Mosh Dataset for the adversarial training in 3D human mesh estimation
+    task.
+
+    The dataset return a dict containing real-world SMPL parameters.
+
+    Args:
+        ann_file (str): Path to the annotation file.
+        pipeline (list[dict | callable]): A sequence of data transforms.
+        test_mode (bool): Store True when building test or
+            validation dataset. Default: False.
+    """
+
+    def __init__(self, ann_file, pipeline, test_mode=False):
+
+        self.annotations_path = ann_file
+        self.pipeline = pipeline
+        self.test_mode = test_mode
+
+        self.db = self._get_db(ann_file)
+        self.pipeline = Compose(self.pipeline)
+
+    def _get_db(self, ann_file):
+        """Load dataset."""
+        data = np.load(ann_file)
+        _betas = data['shape'].astype(np.float32)
+        _poses = data['pose'].astype(np.float32)
+        tmpl = dict(
```

Comment:
maybe revising is not necessary, but it can be shorter

 ```python
gt_db = [dict(pose=p, beta=b) for p, b in zip(_poses, _betas)]
```

Reply:
I prefer to keep the codes here unchanged, because there are many other labels for an item.

Reply:
fine